core/src/banking_stage/committer.rs

```diff
@@ -76,12 +74,6 @@ impl Committer {
         execute_and_commit_timings: &mut LeaderExecuteAndCommitTimings,
         processed_counts: &ProcessedTransactionCounts,
     ) -> (u64, Vec<CommitTransactionDetails>) {
-        let processed_transactions = processing_results
-            .iter()
-            .zip(batch.sanitized_transactions())
-            .filter_map(|(processing_result, tx)| processing_result.was_processed().then_some(tx))
-            .collect_vec();
```

Comment:
wrt my other comment - we can get rid of the `collect_vec` here, but we don't need to get rid of the filtering, right?